lib/GADS.pm

```diff
@@ -3024,6 +3025,99 @@ prefix '/:layout_name' => sub {
 
     };
 
+    any ['get', 'post'] => '/historic_purge/' => require_login sub {
+
+        my $layout = var('layout') or pass;
+
+        my $user        = logged_in_user;
+        my $view        = current_view($user, $layout);
+
+        my $records = GADS::Records->new(
+            user                => $user,
+            layout              => $layout,
+            schema              => schema,
+            include_children    => 1,
+            view_limit_extra_id => undef, # Override any value that may be set
+            view                => $view,
+        );
+
+        my $params = +{};
+        my $record_data = [];
+
+        if(param 'submit') {
+            my @columns = body_parameters->get_all('record');
+            $records->columns([@columns]);
+        } elsif(param 'purge') {
+            my @columns = split(/,/,body_parameters->get('records'));
```

Comment:
This would be better as multiple hidden fields, and then it could be combined with the block above.